Classes/Util.php

```diff
@@ -274,22 +274,23 @@ public static function getConfigurationFromPageId(
 
         if ($initializeTsfe) {
             self::initializeTsfe($pageId, $language);
-            if (!empty($configurationToUse)) {
-                return self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId, $language, $path);
+            if (empty($configurationToUse)) {
+                $configurationToUse = self::getConfigurationFromInitializedTSFE($path);
             }
-            $configurationToUse = self::getConfigurationFromInitializedTSFE($path);
         } else {
-            if (!empty($configurationToUse)) {
-                return self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId, $language, $path);
+            if (empty($configurationToUse)) {
+                $configurationToUse = self::getConfigurationFromExistingTSFE($pageId, $path, $language);
             }
-            $configurationToUse = self::getConfigurationFromExistingTSFE($pageId, $path, $language);
         }
 
         if ($useCache) {
             $cache->set($cacheId, $configurationToUse);
         }
 
-        return $configurationsByCacheKey[$cacheId] = self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId, $language, $path);
+        $configurationsByCacheKey[$cacheId] = self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId,
```

Comment:
These lines do not change any functionality. If a cosmetic change is desired please open a secondary pull request.